assets/src/components/activities/common/hints/authoring/hintActions.ts

```diff
@@ -0,0 +1,38 @@
+import { getHint, getHints } from 'components/activities/common/hints/authoring/hintUtils';
+import { HasHints, Hint, PostUndoable, RichText } from 'components/activities/types';
+
+export const HintActions = {
+  addHint(hint: Hint) {
+    return (model: HasHints, post: PostUndoable) => {
+      // new hints are always cognitive hints. they should be inserted
+      // right before the bottomOut hint at the end of the list
+      const bottomOutIndex = getHints(model).length - 1;
+      getHints(model).splice(bottomOutIndex, 0, hint);
+    };
+  },
+
+  editHint(id: string, content: RichText) {
+    return (model: HasHints, post: PostUndoable) => {
+      getHint(model, id).content = content;
+    };
+  },
+
+  removeHint(id: string) {
+    return (model: HasHints, post: PostUndoable) => {
+      const hint = getHint(model, id);
+      const index = getHints(model).findIndex((h) => h.id === id);
+      model.authoring.parts[0].hints = getHints(model).filter((h) => h.id !== id);
+      post({
+        description: 'Removed a hint',
+        operations: [
+          {
+            path: '$.authoring.parts[0].hints',
```

Comment:
The fact that this is part 0 should not be hardcoded here. In fact, almost none of this path makes sense to hardcode. This will make this component unusable in multi-part activities, and in activities that have non-standard structuress.  An easy way to address this is to allow this path to be passed in to this component as a prop.

Reply:
I'm not sure what the best way to fix this is. Immer makes mutable updates, so the hints actions have to update the property of an object, which means you can't just pass a string path for the `PostUndoable`, you have to have access to the parent object where the hints are stored. So the part ID could be passed as an argument, but these hint actions aren't useable for non-standard structures. 

I could back out these changes and have every activity write its own hint update logic, but I couldn't figure out a way to fix all the duplication that we had without moving to redux (see comment above, this created its own problems).

Reply:
I think the easiest way would be to expose a prop on `HintsAuthoring` called "hintPath" that the MCQ Authoring component would set to `"$.authoring.parts[0].hints"`.  That would have to get passed all the way into the code then that creates and invokes the `post` method.